### ISSUE TITLE
bug: missing root element's scope id as classname in nested child

### ISSUE THREAD

**Issue by yigityuce** [ionic-team/stencil#5702]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.17.1

### Current Behavior

think about having a component structure like below:

```
my-timepicker -------> my-input ---------> input (native)
             |
             | ------> my-menu --------> my-popover ---------> whatever


my-popover (can be used as standalone)

```

given that component structure I have a style for popover (eg.: `padding: 8;`) and I want to customize popover style from the `timepicker` component's style file like:

```scss
:host {
  my-menu {
     my-popover {
        padding: 32px;
     }
  }
}
```

Current behavior: nested element selector is **NOT** working since during the scss compilation phase it adds `.sc-my-timepicker` class to the `my-popover` selector and the compiled CSS file becomes sth like this:

```css
.sc-my-timepicker-h my-menu.sc-my-timepicker my-popover.sc-my-timepicker {
  padding: 32px;
}
```

the problem is root component's scope id (`sc-my-timepicker`) is **NOT** being added to the nested component's classNames. So the selector which is defined in root component is **NOT** working.

### Expected Behavior

works as expected

current:

<img width="712" alt="current" src="https://github.com/ionic-team/stencil/assets/5856230/eabcb0ee-57b4-42e6-a1de-2385f7f33bdf">

expected:

<img width="711" alt="expected" src="https://github.com/ionic-team/stencil/assets/5856230/8e5c3fdd-5c78-48f9-b259-e04836ab77d2">

### System Info

```shell
Your current version of Node is v16.18.0, however Stencil's recommendation is v18.16.0 or greater. Note that future versions of Stencil will eventually remove support for older Node versions and an Active LTS version is recommended (https://nodejs.org/en/about/releases/).


      System: node 16.18.0
    Platform: darwin (23.4.0)
   CPU Model: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 cpus)
    Compiler: /Users/yigit.yuce/Documents/Projects/personal/yigityuce.github/stencil-v4-sc-nested-css-scope-id/node_modules/@stencil/core/compiler/stencil.js
       Build: 1713902307
     Stencil: 4.17.1 🚒
  TypeScript: 5.4.5
      Rollup: 2.56.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.30.3
```

### Steps to Reproduce

https://github.com/yigityuce/stencil-v4-sc-nested-css-scope-id

### Code Reproduction URL

https://github.com/yigityuce/stencil-v4-sc-nested-css-scope-id

**Comment by christian-bromann** [comment]
Thanks for raising this well described bug and providing a reproducible example. I took a look and reviewed the PR.

**Comment by yigityuce** [comment]
@christian-bromann PR is updated

**Comment by yigityuce** [comment]
closing the issue since #5704 is merged 🚀 thanks all!

**Comment by rwaskiewicz** [comment]
The fix for this PR was included in today's [v4.18.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.18.0)
